libpod/runtime_ctr.go

```diff
@@ -47,6 +47,32 @@ func (r *Runtime) NewContainer(ctx context.Context, rSpec *spec.Spec, options ..
 	return r.newContainer(ctx, rSpec, options...)
 }
 
+func (r *Runtime) PrepareVolumeOnCreateContainer(ctx context.Context, ctr *Container) (retErr error) {
+	// Copy the content from the underlying image into the newly created
+	// volume if configured to do so.
+	if !r.config.Containers.PrepareVolumeOnCreate {
+		return nil
+	}
+
+	defer func() {
```

Comment:
Since it's only the mountstorage that can fail, I don't think you need to cleanup storage at all.

Reply:
The `prepare()` function that calls `mountStorage()` does call `cleanupStorage()` afterwards. I am wondering whether not calling `cleanupStorage()` will result in any leak for the containers that are just created but never started/stopped.